### PR TITLE
feat: Adds `eslint` for `ts` and `js` files

### PR DESCRIPTION
**Pull request by tuliomir**
### Acceptance Criteria
- Adds `eslint` configuration for all of the project code
- Adds scripts on the `package.json` for easy linting, formatting and error fixing

### Notes
A `.eslintrc` file already existed in the repository and was being used to lint the integration test files. The rules on this file were preserved, and it was only moved to a `js` structure for improved maintenance convenience.

A newer major version of Eslint is available, but has some incompatibilities with some of our other dependencies. This is being tracked by:
- #661 

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -53,14 +56,21 @@
...
-    "eslint": "8.36.0",
-    "eslint-config-airbnb-base": "15.0.0",
-    "eslint-plugin-import": "2.27.5",
+    "eslint": "8.57.0",
+    "eslint-config-airbnb-base": "^15.0.0",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-import-resolver-typescript": "^3.6.1",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-jest": "^28.6.0",
+    "eslint-plugin-prettier": "^5.1.3",
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply package.json]: Sure! Fixed them.

File: package.json [hunk 2]
```diff
@@ -53,14 +56,21 @@
     "@types/level": "6.0.3",
     "@types/lodash": "4.17.5",
     "@types/node": "20.14.2",
+    "@typescript-eslint/eslint-plugin": "^7.13.0",
+    "@typescript-eslint/parser": "^7.13.0",
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply package.json]: Done :+1:

File: package.json [hunk 3]
```diff
@@ -53,14 +56,21 @@
...
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-import-resolver-typescript": "^3.6.1",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-jest": "^28.6.0",
+    "eslint-plugin-prettier": "^5.1.3",
     "jest": "29.7.0",
     "jest-html-reporter": "3.10.2",
     "jest-localstorage-mock": "2.4.26",
     "mock-socket": "9.3.1",
+    "prettier": "^3.3.2",
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply package.json]: Done

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  ],
+  plugins: ['jest'],
+  parserOptions: {
+    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
+    sourceType: 'module', // Allows for the use of imports
+  },
+  settings: {
+    'import/resolver': {
+      node: {
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply .eslintrc.js]: That's right. Removed those XML versions on the `import/extensions` rule as well.

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+module.exports = {
+  parser: '@typescript-eslint/parser', // Specifies the ESLint parser
+  extends: [
+    'airbnb-base',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
+  ],
+  plugins: ['jest'],
+  parserOptions: {
+    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply .eslintrc.js]: Updated to the latest available: `2023`.
There's an option to always use the `latest`, but I believe it does not fit with the project's stability requirements, so I fixed it on 2023.

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        js: 'never',
+        jsx: 'never',
+        ts: 'never',
+        tsx: 'never',
+      },
+    ],
+  },
+  overrides: [
+    {
+      files: ['*.ts', '*.tsx'],
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply .eslintrc.js]: Fixed

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    'guard-for-in': 'off',
+    'no-bitwise': 'off',
+    'import/extensions': [
+      'error',
+      'ignorePackages',
+      {
+        js: 'never',
+        jsx: 'never',
+        ts: 'never',
+        tsx: 'never',
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply .eslintrc.js]: Done

**Review by andreabadesso — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**